index.js

```diff
@@ -294,6 +294,38 @@ class Statham {
       });
     });
   }
+
+  /**
+   * Search and return keys and values that match given string.
+   *
+   * @param {String} phrase
+   *
+   * @returns {{}}
+   */
+  search(phrase) {
+    let tmp  = {};
+    let data = this.data;
+
+    if (this.isModeNested()) {
+      data = flatten(this.data);
```

Comment:
Perhaps cache this on a property. What if we search multiple times? This could really hit performance.